### ISSUE TITLE
Permission denied error when starting the container with custom user and UID/GID

### ISSUE THREAD

**Issue by hans-permana** [jupyter/docker-stacks#442]
**What docker image you are using?**

`jupyter/base-notebook`

**What complete docker command do you run to launch the container (omitting sensitive values)?**

`docker run -it --user=root -e NB_USER=user -e NB_UID=50000 -e NB_GID=50000 jupyter/base-notebook /usr/local/bin/start-singleuser.sh`

**What steps do you take once the container is running to reproduce the issue?**

Run the docker run command

**What do you expect to happen?**

I do not see any errors and Jupyter server is started.

**What actually happens?**

observed `PermissionError: [Errno 13] Permission denied: '/home/user'`. I believe this is due to the directory of the new user has not been created. The user and the home directory have been changed by [this command](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/start.sh#L10) but the new home directory does not yet exist at that point. Hence, it produced an error when starting `jupyterhub-singleuser` conda command.

Possible solution may be to add `mv /home/jovyan /home/$NB_USER` command after the change of user and its home directory .

**Comment by parente** [comment]
Are you launching jupyterhub-singleuser directly, or are you seeing this when launching these images with dockerspawner and jupyterhub?

**Comment by hans-permana** [comment]
hi @parente, I got this error when launching this image directly. I have also observed an issue at [this line](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/start.sh#L29) when launching it from Jupyterhub (with [swarmspawner](https://github.com/cassinyio/SwarmSpawner)), with the error message `usermod: Failed to change ownership of the home directory`. I did a little bit of debugging and I noticed that `/home/user` was already created, but still, I thought `usermod` should not fail just because of existing directory.
